src/Numerical/HBLAS/BLAS/Internal.hs

```diff
@@ -182,6 +186,35 @@ gemvAbstraction gemvName gemvSafeFFI gemvUnsafeFFI constHandler = gemv
                          (fromIntegral bstride) betaPtr cp (fromIntegral cstride)
 
 
+{-# NOINLINE gerAbstraction #-}
+gerAbstraction :: (SM.Storable el, PrimMonad m)
+               => String
+               -> GerFunFFI el
+               -> GerFunFFI el
+               -> forall orient . GerFun el orient (PrimState m) m
+gerAbstraction gerName gerSafeFFI gerUnsafeFFI = ger
+    where
+      shouldCallFast :: Int -> Int -> Bool
+      shouldCallFast m n = flopsThreshold >= (fromIntegral m :: Int64)
```

Comment:
You're correct. It is correct as written. Fast is true when the product is
less than the threshold.   Probably I should write a little  helper
function to make it less confusing
On Thursday, May 7, 2015, Jueji Yang <notifications@github.com> wrote:

> In src/Numerical/HBLAS/BLAS/Internal.hs
> <https://github.com/wellposed/hblas/pull/18#discussion_r29869978>:
>
> > @@ -182,6 +186,35 @@ gemvAbstraction gemvName gemvSafeFFI gemvUnsafeFFI constHandler = gemv
> >                           (fromIntegral bstride) betaPtr cp (fromIntegral cstride)
> >
> >
> > +{-# NOINLINE gerAbstraction #-}
> > +gerAbstraction :: (SM.Storable el, PrimMonad m)
> > +               => String
> > +               -> GerFunFFI el
> > +               -> GerFunFFI el
> > +               -> forall orient . GerFun el orient (PrimState m) m
> > +gerAbstraction gerName gerSafeFFI gerUnsafeFFI = ger
> > +    where
> > +      shouldCallFast :: Int -> Int -> Bool
> > +      shouldCallFast m n = flopsThreshold >= (fromIntegral m :: Int64)
>
> Yes, I think we should be calling the fast code IF m*n is *LESS*(or *LEQ*)
> than flopsThreshold. But I think it is the same that we should be calling
> the fast code IF flopsThreshold is *GREATER*(or *GEQ*) than m*n?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wellposed/hblas/pull/18/files#r29869978>.
>